Automated releases for assets

rogeruiz opened the pull request:
@maya and @ericadeahl touched on the need for automated release ZIP archives for the asset files when discussing the new update designs 18F/web-design-standards#1274.

This patch series aims to automate the creation and distribution of the asset files into ZIP archives separated by file type.
- omnigraffle
- eps
- ai/ase
- sketch / sketchpalette

This update automated release process introduces some new changes to the workflow for this repo. Most notably, the workflow now incorporates `gulp` and `npm run` commands.
- `npm install` will install the dependencies for the automation process and smoke tests.
- `npm test:extensions` will verify that the support file extensions ☝️ exist in the repository.
- `npm release` will generate four types of zip archives based on the supported file extensions. _note the version number is dictated by the version number in the package.
  1. `uswds-assets-0.9.4-ai.zip`
  2. `uswds-assets-0.9.4-eps.zip`
  3. `uswds-assets-0.9.4-omnigraffle.zip`
  4. `uswds-assets-0.9.4-sketch.zip`

This workflow introduces a new versioning system for this repository. In order to stay in sync with the standards documentation site, this repo should mirror the versioning in `18F/web-design-standards`. This will require tagging and publishing those tags to Github. Any changes that were introduced in the standards in code, should be reflected in the design files if necessary. Currently, there is no automated way to do this and it should be a manual process since the design files will need to be evaluated. If there are no changes between versions, then the version number is bumped and published just to remain in sync with the code package.

rogeruiz committed:
Update ignored paths

rogeruiz committed:
Add automated release archives and tests

rogeruiz committed:
Show colors after installing

juliasolorzano commented:
LGTM and thanks for knocking this out @rogeruiz! Let's add some documentation in the `README.md` around this process too.

yozlet commented on config/gulp/release.js:
```diff
@@ -0,0 +1,149 @@
+
+});
+
+gulp.task('release:process', function () {
```
Hey @rogeruiz, can you explain this task? I'm having a hard time understanding what it's doing.

rogeruiz replied:
@yozlet Sure! The `release:process` task takes each type of file from the given directories / extensions and splits them out into separate `gulp#streams`. This is necessary because the source directories all contain every file in a particular folder, [but for the upcoming version of the `uswds` documentation site a separate ZIP archive is referenced for each `filetype`](https://standards-staging-new-design.apps.cloud.gov/download/). So, this task merges all these source streams and generates a folder for each `extension` type. See [this line](https://github.com/18F/web-design-standards-assets/pull/28/files#diff-09fe040bc48f79929c344efe0d53508fR5) for the start of the file extensions that are supported by this task.

The zip files and folders generated by this build process look like this:

![screen shot 2016-06-28 at 2 18 26 pm](https://cloud.githubusercontent.com/assets/706004/16427417/3b73b41e-3d3b-11e6-8e97-ed747fef0430.png)

_On a related note, I think it makes more sense to comment these `gulp.task` calls a bit better and I'll update the PR._

maya commented:
When this is complete, we'll need to update the download links here: https://github.com/18F/web-design-standards/blob/add-new-designs/docs/pages/getting-started/download.html

rogeruiz commented:
A direct link to the include mentioned above:
https://github.com/18F/web-design-standards/blob/add-new-designs/docs/_includes/download-buttons-design.html

maya commented:
Thanks @rogeruiz! I moved it out to an include this afternoon as we're using it in two places: Downloads page and For Designers page.

maya commented:
Just curious, why are we adding a version number to these files? cc: @ericadeahl @rogeruiz

rogeruiz committed:
Add some documentation comments around gulp tasks

rogeruiz committed:
Clean up whitespace

rogeruiz committed:
Add documentation around automation

rogeruiz committed:
Remove all aspects of versioning

rogeruiz committed:
Update notify image path

rogeruiz commented:
Alrighty, I've updated the documentation here to include some instructions around running the gulp commands. I also went ahead and removed the versioning aspect from the project as well.

Below is a screenshot of the zip archives that this creates: 🎉 

![screen shot 2016-06-29 at 1 13 17 pm](https://cloud.githubusercontent.com/assets/706004/16461624/71034b7e-3dfb-11e6-8355-88d19de030da.png)

Commit:
Copy edits

A few copy edits (capitalization + passive > active voice).

Commit:
Merge pull request #33 from 18F/kategarklavs-patch-2

Copy edits

juliasolorzano commented:
LGTM - nice work (as always) @rogeruiz! ^5
